clientapi/routing/register.go

```diff
@@ -88,28 +93,75 @@ func (d *sessionsDict) GetCompletedStages(sessionID string) []authtypes.LoginTyp
 	return make([]authtypes.LoginType, 0)
 }
 
+// addParams adds a registerRequest to a sessionID and starts a timer to delete that registerRequest
+func (d *sessionsDict) addParams(sessionID string, r registerRequest) {
+	d.Lock()
+	d.Unlock()
+	d.startTimer(defaultTimeOut, sessionID)
+	d.params[sessionID] = r
+}
+
+func (d *sessionsDict) getParams(sessionID string) (registerRequest, bool) {
+	d.RLock()
+	defer d.RUnlock()
+	r, ok := d.params[sessionID]
+	return r, ok
+}
+
+// deleteSession cleans up a given session, either because the registration completed
+// successfully, or because a given timeout (default: 5min) was reached.
+func (d *sessionsDict) deleteSession(sessionID string) {
+	d.Lock()
+	defer d.Unlock()
+	delete(d.params, sessionID)
+	delete(d.sessions, sessionID)
+	// stop the timer, e.g. because the registration was completed
+	if t, ok := d.timer[sessionID]; ok {
+		t.Stop()
```

Comment:
See notes at https://pkg.go.dev/time#Timer.Stop — you need to make sure the channel is drained.

Reply:
Changed to `time.AfterFunc`, but at one point it just deadlocks when trying to drain the channel. It's commented in the code.

Reply:
```
if !t.Stop() {
    select {
    case <-t.C:
    default:
     }
}
```
... should solve it.